### PR TITLE
Suggestions interface

### PR DESCRIPTION
**Pull request by diogoscf**
I have deconflicted the "suggestions-Kira-D" branch and master, while still making the interface work. (I am from Google Code-In)

### CODE REVIEW

**Comment by shardulc** [comment]
@diogoscf Can you fix all the things Hound is complaining about, and make sure the CircleCI build passes?

**Comment by sushain97** [comment]
In addition to fixing the Hound errors, please remove the `.idea/` files from this PR. I suggest adding them to your global `.gitignore`.

File: assets/js/translator.js [hunk 1]
```diff
@@ -6,12 +6,15 @@ var curSrcLang, curDstLang;
 var recentSrcLangs = [], recentDstLangs = [];
 var droppedFile;
 var translateRequest;
+var grecaptcha;
+var recaptchaRenderCallback;
```
[the remark this answers is not in the document]
Comment (diogoscf) [reply assets/js/translator.js]: It is used by the google recaptcha script. Anyone know how to fix this?

**Comment by diogoscf** [comment]
`recaptchaRenderCallback` is used by the google recaptcha script, so I can't remove it. How do I get circleci and the Hound to stop saying it's an error?

**Comment by diogoscf** [comment]
`recaptchaRenderCallback` is used by the google recaptcha script, so I can't remove it. How do I get circleci and the Hound to stop saying it's an error? @shardulc @sushain97

**Comment by sushain97** [comment]
@diogoscf https://stackoverflow.com/a/29592334/1266600

File: .gitignore [hunk 2]
```diff
@@ -12,3 +12,6 @@ assets/js/config.js
 .Trashes
 Thumbs.db
 ehthumbs.db
+
+#Idea
```
Comment (sushain97) [inline .gitignore]: Please don't modify the project's `.gitignore` by adding editor/environment specific files. Those belong in your global `.gitignore`. I think this has been violated a little bit above but we shouldn't make it worse.

File: assets/js/util.js [hunk 3]
```diff
@@ -413,3 +413,19 @@ function displayInstallationNotification() {
 /*:: import {persistChoices} from "./persistence.js" */
 /*:: import {iso639Codes, iso639CodesInverse} from "./localization.js" */
 /*:: import {populateTranslationList, showTranslateWebpageInterface} from "./translator.js" */
+
+/* eslint-disable */
+String.prototype.hashCode = function () {
```
Comment (sushain97) [inline assets/js/util.js]: This shouldn't be after the exports.

**Comment by sushain97** [comment]
@diogoscf could you provide some screenshots/videos as requested on the task?

**Comment by diogoscf** [comment]
I have provided screenshots on the google code-in website

**Comment by diogoscf** [comment]
I have provided screenshots on the google code-in website @sushain97

**Comment by sushain97** [comment]
@jonorthwash it's time for us to re-look at this and come up with a concrete list of suggestions for improvements to be made on this suggestions interface :)

@shardulc feel free to also offer ideas

Once we have a concrete list, we can make a GCI task.

**Comment by jonorthwash** [comment]
What's the current state of it?

**Comment by sushain97** [comment]
Check out the screenshots on the GCI task.

**Comment by jonorthwash** [comment]
>  the GCI task.

Er, do you have a url?

**Comment by sushain97** [comment]
@jonorthwash https://codein.withgoogle.com/dashboard/task-instances/4618273364639744/

**Comment by jonorthwash** [comment]
Does it actually submit the text?

**Comment by shardulc** [comment]
It cannot, because there is no token for the captcha.

**Comment by jonorthwash** [comment]
Is that something that needs to be fixed, then?

**Comment by shardulc** [comment]
Yes, we will likely need a captcha token for testing; I guess we should generate one for Apertium and give it to whoever decides to work on this task?

**Comment by sushain97** [comment]
@shardulc @jonorthwash I went ahead and set it up on my server (with a recaptcha key) so we can easily test: https://azurite.skc.name/apertium-html-tools/build/index.eng.html

You need to check `Mark unknown words` for it to take effect (type some nonsense to translate).

There are definitely some UI touches that need to take place but I haven't thoroughly tested it yet. Whoever can get to it first should take a crack at it!

I also have a hunch that it's making my CPU spike... Here be dragons.

**Comment by shardulc** [comment]
@sushain97 Should we merge this into `goavki/suggestions` and then discuss further on a new issue?

**Comment by sushain97** [comment]
> @sushain97 Should we merge this into goavki/suggestions and then discuss further on a new issue?

I think that's a good idea.

**Pull request merged**